app/models.py

```diff
@@ -791,6 +791,137 @@ def serialize(self, data=None):
         return supplier_framework
 
 
+class Application(db.Model):
```

Comment:
If you specify the relationship with a string it will defer the lookup: 

application = db.relationship('Application' ....

and then you can move application back to where it was